walletunlocker/service.go

```diff
@@ -100,25 +117,43 @@ type UnlockerService struct {
 	// sent.
 	UnlockMsgs chan *WalletUnlockMsg
 
+	// MacResponseChan is the channel for sending back the admin macaroon to
+	// the WalletUnlocker service.
+	MacResponseChan chan []byte
+
 	chainDir       string
 	noFreelistSync bool
 	netParams      *chaincfg.Params
-	macaroonFiles  []string
+
+	// macaroonFiles is the path to the three generated macaroons with
+	// different access permissions. These might not exist in a stateless
+	// initialization of lnd.
+	macaroonFiles []string
 }
 
 // New creates and returns a new UnlockerService.
 func New(chainDir string, params *chaincfg.Params, noFreelistSync bool,
 	macaroonFiles []string) *UnlockerService {
 
 	return &UnlockerService{
-		InitMsgs:      make(chan *WalletInitMsg, 1),
-		UnlockMsgs:    make(chan *WalletUnlockMsg, 1),
-		chainDir:      chainDir,
-		netParams:     params,
-		macaroonFiles: macaroonFiles,
+		InitMsgs:   make(chan *WalletInitMsg, 1),
+		UnlockMsgs: make(chan *WalletUnlockMsg, 1),
+
+		// Make sure we buffer the channel is buffered so the main lnd
+		// goroutine isn't blocking on writing to it.
+		MacResponseChan: make(chan []byte, 1),
+		chainDir:        chainDir,
+		netParams:       params,
+		macaroonFiles:   macaroonFiles,
 	}
 }
 
+// Stop closes the macaroon response channel to make sure all pending requests
+// are unblocked in case no macaroon was sent back to the unlocker yet.
+func (u *UnlockerService) Stop() {
```

Comment:
instead of having this stop method, maybe clearer to close the channel directly from the caller?